conda-store-server/conda_store_server/worker/tasks.py

```diff
@@ -254,14 +255,20 @@ def task_delete_build(self, build_id):
 
         build = api.get_build(db, build_id)
 
+        # Deletes build artifacts for this build
         conda_store.log.info(f"deleting artifacts for build={build.id}")
         for build_artifact in api.list_build_artifacts(
             db,
             build_id=build_id,
             excluded_artifact_types=settings.build_artifacts_kept_on_deletion,
         ).all():
             delete_build_artifact(db, conda_store, build_artifact)
-        conda_store.db.commit()
+
+        # Updates build size and marks build as deleted
+        build.deleted_on = datetime.datetime.utcnow()
+        build.size = 0
```

Comment:
Metadata fix: size value of the current build is shown in the web UI as `size`. Need to update that when deleting artifacts here.

How it works in more detail:

```html
{% for environment in environments %}
<div class="card my-2">
    <div class="card-body">
        <h5 class="card-title">
            <a href="{{ url_for('ui_get_environment', namespace=environment.namespace.name, environment_name=environment.name) }}">{{ environment.namespace.name }} / <span class="badge badge-info">{{ environment.name }}</span></a>
            <span class="badge badge-light">{{ (environment.current_build.size or 0) | filesizeformat(true) }}</span>
        </h5>
```

=>

```html
 document.querySelectorAll('button[data-action="delete"]').forEach(item => {
     item.addEventListener('click', (event) => deleteBuild(event));
 })
```

=>

```html
 async function deleteBuild(event) {
...
     let url = `{{ url_for('api_list_builds') }}${ buildId }/`;
     let response = await fetch(url, { method: 'DELETE' });
...
```

=>

```py
@router_api.delete(
    "/build/{build_id}/",
    response_model=schema.APIAckResponse,
)
async def api_delete_build(
...
        conda_store.delete_build(db, build_id)
```

=>

```py
     def delete_build(self, db: Session, build_id: int):
...
        tasks.task_delete_build.si(build.id).apply_async()
```

=>

```py
task_delete_build (this function)
```